### PR TITLE
Add sequence numbers to cat shards API

### PR DESCRIPTION
**Pull request by jasontedor**
This commit adds the max sequence number and local and global
checkpoints to the cat shards API.

### CODE REVIEW

**Comment by jasontedor** [comment]
This gives output like:

```
15:50:54 [jason:~] $ curl -XGET "localhost:9200/_cat/shards?h=index,shard,prirep,docs,seq_no.*&v"
index shard prirep docs seq_no.max seq_no.local_checkpoint seq_no.global_checkpoint
i     4     p       111        221                     221                      221
i     4     r       111        221                     221                      221
i     1     r        97        193                     193                      193
i     1     p        97        193                     193                      193
i     2     p       108        215                     215                      215
i     2     r       108        215                     215                      215
i     3     r        98        195                     195                      195
i     3     p        98        195                     195                      195
i     0     p        98        195                     195                      195
i     0     r        98        195                     195                      195
```

**Comment by bleskes** [comment]
LGTM. Do you use it for debugging?

File: core/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java [hunk 1]
```diff
@@ -295,6 +299,10 @@ private Table buildTable(RestRequest request, ClusterStateResponse state, Indice
             table.addCell(commonStats == null ? null : commonStats.getSegments().getVersionMapMemory());
             table.addCell(commonStats == null ? null : commonStats.getSegments().getBitsetMemory());
 
+            table.addCell(shardStats == null || shardStats.getSeqNoStats() == null ? null : shardStats.getSeqNoStats().getMaxSeqNo());
+            table.addCell(shardStats == null || shardStats.getSeqNoStats() == null ? null : shardStats.getSeqNoStats().getLocalCheckpoint());
+            table.addCell(commitStats == null || shardStats.getSeqNoStats() == null ? null : shardStats.getSeqNoStats().getMaxSeqNo());
```
Comment (tlrx) [inline core/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java]: Shouldn't this be `getGlobalCheckpoint()`?

**Comment by tlrx** [comment]
Left a comment - otherwise LGTM

File: core/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java (hunk 1 above)
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java]: Yes, thanks for catching @tlrx! I pushed 0f94e40.

**Comment by jasontedor** [comment]
> Do you use it for debugging?

@bleskes Yes, I just find it easier than filtering the indices stats API.

**Pull request merged**